### PR TITLE
Fix reentrancy bug in io_uring transport implementation related to

### PR DESCRIPTION
**Pull request by normanmaurer**
writes

Motivation:

We need to carefully manage state in terms of writing to guard against
rentrancy problems that could lead to corrupt state in the
ChannelOutboundBuffer

Modifications:

Only reset the flag once removeBytes(...) was called

Result:

No more reentrancy bug related to writes.

### CODE REVIEW

**Comment by normanmaurer** [comment]
This caused some flaky tests.

File: transport-native-io_uring/src/main/java/io/netty/channel/uring/AbstractIOUringChannel.java [hunk 1]
```diff
@@ -315,11 +315,8 @@ private void doWriteMultiple(ChannelOutboundBuffer in) {
                  // This should never happem, anyway fallback to single write.
                  doWriteSingle((ByteBuf) in.current());
              }
-
-             if (iovecArray.count() > 0) {
```
Comment (njhill) [inline transport-native-io_uring/src/main/java/io/netty/channel/uring/AbstractIOUringChannel.java:319]: because count will always be > 1 here?

**Review by njhill — APPROVED** [review]
Makes me think of a possibly optimization for writing/flushing ... to defer processing the writes (flushes) until _submission_ time rather than scheduling time. A placeholder SQE could still be put in the queue but filled in when the actual submit happens. Thus mutliple flushes would be coalesced where possible.

File: transport-native-io_uring/src/main/java/io/netty/channel/uring/AbstractIOUringChannel.java (hunk 1 above)
Comment (normanmaurer) [reply transport-native-io_uring/src/main/java/io/netty/channel/uring/AbstractIOUringChannel.java:319]: yep

**Comment by normanmaurer** [comment]
@njhill lets save this for later

**Pull request merged**